### DISCUSSION TITLE
Action and option to declutter type hints

### DISCUSSION THREAD

**Discussion by rik** [microsoft/pylance-release#4723] [Enhancement]
As useful as type hints can be, they can sometimes be visually overwhelming. Just like imports, it would be nice to hide them sometimes.

- Action to show/hide type hints.
- Option to decide if they should be shown/hidden when first opening a file.

[PyCharm shipped a similar feature in 2023.2](https://www.jetbrains.com/pycharm/whatsnew/#:~:text=Option%20to%20collapse%20Python%20type%20annotations).

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by bschnurr** [comment]
checkout the editor command to hide hints until a key is pressed `CTRL + ALT`
![image](https://github.com/microsoft/pylance-release/assets/1946977/50a0e5b0-4241-4659-9091-0ad2cba32431)

**Reply by rik** [reply]
This enhancement request is about type hints that are in the source code, not inlay hints that are inferred.
